src/PyDynamic/uncertainty/propagate_convolution.py

```diff
@@ -167,3 +162,18 @@ def convolve_unc(x1, U1, x2, U2, mode="full"):
         raise ValueError(f'convolve_unc: Mode "{mode}" is not supported.')
 
     return conv, Uconv
+
+
+def ensure_cov_matrix(unc_array):
+    """
+    Converts 1D-arrays of standard uncertainties into the corresponding 
+    (diagonal) covariance matrix by *square*+diag. 
+
+    Does not modify inputs which are 2D-arrays or None.
+    """
+    
+    # squeeze() ensures proper execution on arrays with only one dimension of non-zero length
+    if isinstance(unc_array, np.ndarray) and len(unc_array.squeeze().shape) == 1:
```

Comment:
I'd suggest
```suggestion
    if unc_array is not None and len(unc_array.squeeze().shape) == 1:
```
to better reflect the actual meaning of the first condition. For the second I am wondering if we want to include the intermediate squeeze into `misc.tools.is_vector()` and use it here.

Reply:
Good point, I followed your suggestion. 

For now, I renamed the `ensure_cov_matrix` to `_ensure_cov_matrix` to mark it as private.

Reply:
What do you think about `is_vector()`?

Reply:
Resolved by refactoring `is_vector()`.

Reply:
I have to reopen this discussion. Refactoring the `ndarray.squeeze()` into `is_vector` broke other parts of PyDynamic which relied on checking for the shape to be `(n,)` and not `(1,n)`, `(n,1)` or `(n,)`. Part of me still is hesitant to merge this squeeze idea. We would enable the aforementioned maybe more exotic input shapes, but do we really want to forbid one elemental cov matrices?

The `squeeze()` condidition would return `False` for one-elemental NumPy ndarrays:

```ipython
In [1]: import numpy as np
In [2]: array = np.array([1])
Out[2]: array([1])
In [3]: np.array([1]).shape
Out[3]: (1,)
In [4]: np.array([1]).squeeze().shape
Out[4]: ()
In [5]: len(np.array([1]).squeeze().shape)
Out[5]: 0
```

Reply:
I would then opt to remove the (in the context of this PR definitely useful) addition of squeeze and make it very clear from the docstring, what the function expects.

Reply:
As discussed in our code review today, this is what we will do. Thanks for the thoughts and time.